syntax = "proto3";

message BasePair {
    Coordinates start = 1;
    Coordinates end = 2;
}

message Solution {
    repeated BasePair instance = 1;
    repeated int32 order = 2;
    bool success = 3;
    float measure = 4;
    int32 nRouted = 5;
    repeated Path pathData = 6;
    repeated int32 failedConnections = 7;
}

message Path {
    repeated Coordinates pathNodes = 1;
}

message Coordinates {
    int32 x = 1;
    int32 y = 2;
}